# My First Project: Apps Attracts users the most:

In this project I will:

>* preparing code and algorythms for further analizing 



>* analyzing data to help our developers understand what type of apps are likely to attract more users. 

In [1]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv', encoding='utf8')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv', encoding='utf8')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

#print('ios =', ios)

#print('android = ', android_header)

In [2]:
def explore_data(dataset, start, end, row_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') # empty line for better visibility 
        
    if row_and_columns:
        print('number of rows =', len(dataset))
        print('number of columns =', len(dataset[0]))
        
        
        
        
print('android_column names = ', android_header)
print("\n")
print('few rows of android datastet = ')
print("\n")
explore_data(android, 0, 3, True)
print('\n')
print('\n')
print('ios_column names =', ios_header)
print('\n')
print('few rows of ios dataset = ')
print("\n")
explore_data(ios, 0, 3, True)



android_column names =  ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


few rows of android datastet = 


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


number of rows = 10841
number of columns = 13




ios_column names = ['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime

##### As we can see, the column names for Android apps is:
>android_column names =  ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']

###### We can see that all the categories are understandable. But, If we want to know more about them, all we have to do is visit: [Documentation](https://www.kaggle.com/lava18/google-play-store-apps). All categories are described in "columns" block on this page. 

---

##### As we can see, the column names for Ios apps is:
>ios_column names = ['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']

###### We can see that not all the categories are understandable. If we want to know more about the meaning of them, all we have to do is visit: [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps).  All categories are described in "columns" block on this page. 

---
###### Some of categories in Ios and Android apps can be handy for further investigation
---

In [3]:
print('potential_error_in_Android_Dataset=', android[10471:10473])

potential_error_in_Android_Dataset= [['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up'], ['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']]


In [4]:
for row in android:
    if len(row) != len(android_header):
        print('row =',row)
        print("android.index =", android.index(row))

row = ['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
android.index = 10472


**As we can see above, we was right about that kind of error. What's more, That's the only error of that type that exist in our dataset.**

Now, all we have to do is remove this row from our dataset and check 

In [5]:
print("android_10472=", android[10472])
print("\n")
print("android lenth before cleaning 1 =", len(android))
print("\n")
del android[10472]
print("android_10472_after_deletion_is:", android[10472])
print("\n")
print('android lenth after cleaning 1 =', len(android))

android_10472= ['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


android lenth before cleaning 1 = 10841


android_10472_after_deletion_is: ['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


android lenth after cleaning 1 = 10840


**As we can see above, we succeed with deletion of row[10472].
Now it's deleted and after this process, we have new 10472 row.**

## Part 2: deeper filtration:

**After deeper analising of our two data sets, we realized that there are some duplicates in Android apps data. Because of that, firstly we want to show some examples of this issiue, then, we will filer them out of our dataset **

In [6]:
for app in android:
    name =  app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


**As we can see, Instagram appears 4 times in our database. Because of that, we need to filter duplicated data and prepare new, better filtrated database.**

In [7]:
duplicated_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicated_apps.append(name)
    else: 
        unique_apps.append(name)
        
print('number of duplicated apps =',len(duplicated_apps))
print('\n')
print('exaples of duplicated apps =', duplicated_apps[:6])

number of duplicated apps = 1181


exaples of duplicated apps = ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box']


**At this point, there is one more problem. As we saw above in "Instagram" example, the probable reason why some apps are duplicated is related to new updated registry that contains more reviews ( without deleted old one). In this case, we want to preserve the highest(newest) registry. For this case, we need to proceed further filtration.**

**For the starting point, there is a need to count how many uniqe apps names lasted**

In [8]:
print('lenth of unique apps =', len(unique_apps))

lenth of unique apps = 9659


### further filtration:

**First we need to prepare dictionary that will store only the rows that incloude the highest reviews rating. In this manner we discard duplicated entries. **

In [9]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

**At this point, we need to find out if our new dictionary "reviews_max" has the same number of entries as main database minus duplicated registry (1181).

In [10]:
print('android lenth without duplicates =', len(android) - 1181)
print('\n')
print("reviews_max =",len(reviews_max))
print('\n')


android lenth without duplicates = 9659


reviews_max = 9659




After confirmation that both are the same, we can use our dictionary for further filtration.

On this step, we will create two lists:
* one that will store clean data, called: android_clean
* one that will store not duplicated names ( so, when duplicated name occur, it will be discard). This list will be called: already_added

In [11]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
        
#print("android_clean =", android_clean[:])
print('\n')

#print("already_added =", already_added[:])


** After this code preparation, all we have to do is to check: is the new list has proper number of rows **

In [12]:
explore_data(android_clean, 0, 3, True)
#coś jest nie tak....nie wiem co ...

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


number of rows = 9659
number of columns = 13


**last thing we have to do is to rid off apps that are non-English. If an app name contains a character that is greater than 127, then it probably means that the app has a non-English name.**

**The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the ASCII (American Standard Code for Information Interchange) system.**

**We can get the corresponding number of each character using the ord() built-in function.**

In [13]:
def is_english(string):
    for character in string:
        if ord(character) > 127:
            return False
        
    return True
        
        
print(is_english('instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))


True
False
False
False


**After creating the definition of pure-English apps, we discover, that our new definicion isn't fine enugh. Some characters like: "™" and "😜" are traited as non-English. To fix it we have to develop our definition further. All English apps with up to three emoji or other special characters will still be labeled as English.**

In [14]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
            
    if non_ascii > 3:
        return False
    else:
        return True
            
            
print(is_english("instagram"))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
False
True
True


In [15]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
print('explore android 0-3 =') 
explore_data(android_english, 0, 3, True)
print('\n')

print('explore ios 0-3 =') 
explore_data(ios_english, 0, 3, True)

explore android 0-3 =
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


number of rows = 9614
number of columns = 13


explore ios 0-3 =
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '213080

**We can see that we're left with 9614 Android apps and 6183 iOS apps.**

---

** Now We isolate free aps in android and in Ios. 

In [16]:
print(android_header)
print('\n')
print(ios_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


** As we can see price is located in 7th column in Android apps and 4th in Ios apps.

In [17]:
free_android_english = []
free_ios_english = []

for price in android_english:
    free = price[7]
    if free == '0':
        free_android_english.append(price)
    
#print(free_android_english[0:2])

for price in ios_english:
    free = price[4]
    if free == '0.0':
        free_ios_english.append(price)
        
print(len(free_android_english))
print(len(free_ios_english))
    

8864
3222


**We're left with 8864 Android apps and 3222 iOS apps, which should be enough for our analysis.**

---

** At this point our goal is to detect the best apps that exist on both platforms.**

**We can point three main points in our further analyse:**

*To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

*Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we develop it further.

*If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

---
**Next thing needed to do is: Inspect both data sets and identify the columns you could use to generate frequency tables to find out what are the most common genres in each market

In [18]:
print('android =', android_header)
print('\n')
print('ios =', ios_header)

android = ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


ios = ['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


** *For Android genre column is number 9**

** *For Ios genre column is number 11**

**Our conclusion was that we'll need to build a frequency table for the prime_genre column of the App Store data set, and for the Genres and Category columns of the Google Play data set.**

In [19]:
def freq_table(dataset, index):

    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = str(row[index])
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
            
    table_percentages = {}      
    
    for key in table:
            percentage = table[key] / total * 100
            table_percentages[key] = percentage
    return table_percentages
        
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (round(float(table[key]), 2), key) # I added round 2 digits
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0], str("%"))

####Part  Three:

**Let's analyze the frequency table we generated for the prime_genre column of the App Store data set.

In [20]:
print("display_table_for_Ios =")
display_table(free_ios_english, 11)

display_table_for_Ios =
Games : 58.16 %
Entertainment : 7.88 %
Photo & Video : 4.97 %
Education : 3.66 %
Social Networking : 3.29 %
Shopping : 2.61 %
Utilities : 2.51 %
Sports : 2.14 %
Music : 2.05 %
Health & Fitness : 2.02 %
Productivity : 1.74 %
Lifestyle : 1.58 %
News : 1.33 %
Travel : 1.24 %
Finance : 1.12 %
Weather : 0.87 %
Food & Drink : 0.81 %
Reference : 0.56 %
Business : 0.53 %
Book : 0.43 %
Navigation : 0.19 %
Medical : 0.19 %
Catalogs : 0.12 %


**The most common genre is games. The runner-up are:**

    *Entertainment
    *Photo & Video
    *Education
    *Social Networking
    *Shopping
    *Utilities
    *Sports
    *Music
    *Health & Fitness
    
**Other patterns we can see here are:**
* there are 4 general grups with similar frequency
* First - the higher in frequency rate - grup isn't related to work.


**From general impression, we can say that any form of entertainment app is a good idea to be created as a recommended first choice to done on this platform. 

---
**Now let's do the same analyze for Android apps.

In [21]:
print("display_table_for_Android =")
display_table(free_android_english, 9)

display_table_for_Android =
Tools : 8.45 %
Entertainment : 6.07 %
Education : 5.35 %
Business : 4.59 %
Productivity : 3.89 %
Lifestyle : 3.89 %
Finance : 3.7 %
Medical : 3.53 %
Sports : 3.46 %
Personalization : 3.32 %
Communication : 3.24 %
Action : 3.1 %
Health & Fitness : 3.08 %
Photography : 2.94 %
News & Magazines : 2.8 %
Social : 2.66 %
Travel & Local : 2.32 %
Shopping : 2.25 %
Books & Reference : 2.14 %
Simulation : 2.04 %
Dating : 1.86 %
Arcade : 1.85 %
Video Players & Editors : 1.77 %
Casual : 1.76 %
Maps & Navigation : 1.4 %
Food & Drink : 1.24 %
Puzzle : 1.13 %
Racing : 0.99 %
Role Playing : 0.94 %
Libraries & Demo : 0.94 %
Auto & Vehicles : 0.93 %
Strategy : 0.91 %
House & Home : 0.82 %
Weather : 0.8 %
Events : 0.71 %
Adventure : 0.68 %
Comics : 0.61 %
Beauty : 0.6 %
Art & Design : 0.6 %
Parenting : 0.5 %
Card : 0.45 %
Casino : 0.43 %
Trivia : 0.42 %
Educational;Education : 0.39 %
Board : 0.38 %
Educational : 0.37 %
Education;Education : 0.34 %
Word : 0.26 %
Casual;Pretend P

**The most common genre is tools and - before that -games. The runner-up are:**

    *Education
    *Business
    *Productivity
    *Lifestyle
    *Finance
    
**Other patterns we can see here are:**
* It seems to be, that android users mostly works and learn with android apps in comparison to Ios users.
* First - the higher in frequency rate - grup is related to work.


**From general impression, we can say that any form of tool usefull for work is a good idea to be created as a recommended first choice to done on this platform. At this point, we can, also say, that games and entertainment apps is good choice here, 

**The frequency tables we analyzed on the previous screen showed us that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and fun apps. Now, we'd like to get an idea about the kind of apps with the most users.**

**One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.**

**Let's start with calculating the average number of user ratings per app genre on the App Store. To do that, we'll need to:**

> 1. Isolate the apps of each genre.
> 2. Sum up the user ratings for the apps of that genre.
> 3. Divide the sum by the number of apps belonging to that genre (not by the total number of apps).


In [22]:
genre_ios = freq_table(free_ios_english, -5)

print(genre_ios)

{'Social Networking': 3.2898820608317814, 'Photo & Video': 4.9658597144630665, 'Games': 58.16263190564867, 'Music': 2.0484171322160147, 'Reference': 0.5586592178770949, 'Health & Fitness': 2.0173805090006205, 'Weather': 0.8690254500310366, 'Utilities': 2.5139664804469275, 'Travel': 1.2414649286157666, 'Shopping': 2.60707635009311, 'News': 1.3345747982619491, 'Navigation': 0.186219739292365, 'Lifestyle': 1.5828677839851024, 'Entertainment': 7.883302296710118, 'Food & Drink': 0.8069522036002483, 'Sports': 2.1415270018621975, 'Book': 0.4345127250155183, 'Finance': 1.1173184357541899, 'Education': 3.662321539416512, 'Productivity': 1.7380509000620732, 'Business': 0.5276225946617008, 'Catalogs': 0.12414649286157665, 'Medical': 0.186219739292365}


In [23]:
for genre in genre_ios:
        total = 0
        len_genre = 0
        
        for app in free_ios_english:
            genre_app = app[-5]
            if genre_app == genre:
                n_ratings = float(app[5])
                total += n_ratings
                len_genre += 1
        avg_n_ratings = total / len_genre
        print(genre, ':', avg_n_ratings)


Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


** As we can see, Navigation has the most ratings. Business, Reference and Social Networking aren't far from too. **

---
** Now let's calculate the average number of installs per app genre for the Google Play data set. We'll need to use a nested loop, just like in the previous screen. **

In [24]:
genre_android = freq_table(free_android_english, 1)
print(genre_android)

{'ART_AND_DESIGN': 0.6430505415162455, 'AUTO_AND_VEHICLES': 0.9250902527075812, 'BEAUTY': 0.5979241877256317, 'BOOKS_AND_REFERENCE': 2.1435018050541514, 'BUSINESS': 4.591606498194946, 'COMICS': 0.6204873646209386, 'COMMUNICATION': 3.2378158844765346, 'DATING': 1.861462093862816, 'EDUCATION': 1.1620036101083033, 'ENTERTAINMENT': 0.9589350180505415, 'EVENTS': 0.7107400722021661, 'FINANCE': 3.7003610108303246, 'FOOD_AND_DRINK': 1.2409747292418771, 'HEALTH_AND_FITNESS': 3.0798736462093865, 'HOUSE_AND_HOME': 0.8235559566787004, 'LIBRARIES_AND_DEMO': 0.9363718411552346, 'LIFESTYLE': 3.9034296028880866, 'GAME': 9.724729241877256, 'FAMILY': 18.907942238267147, 'MEDICAL': 3.531137184115524, 'SOCIAL': 2.6624548736462095, 'SHOPPING': 2.2450361010830324, 'PHOTOGRAPHY': 2.944494584837545, 'SPORTS': 3.395758122743682, 'TRAVEL_AND_LOCAL': 2.33528880866426, 'TOOLS': 8.461191335740072, 'PERSONALIZATION': 3.3167870036101084, 'PRODUCTIVITY': 3.892148014440433, 'PARENTING': 0.6543321299638989, 'WEATHER': 

In [25]:
for category in genre_android:
    total = 0
    len_category = 0
        
    for row in free_android_english:
        category_app = row[1]
        if category_app == category:
            total_1 = row[5]
            total_1 = total_1.replace('+', '')
            total_1 = total_1.replace(',', '')
            total += float(total_1)
            len_category += 1

    avg_android_installs =  total / len_category
    print(category, ':', avg_android_installs)


ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

** Based on the number of user ratings, we can see that on average, communication apps are ones mostly installed. **